provider/resource_keycloak_realm_keystore_rsa.go

```diff
@@ -0,0 +1,221 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+	"log"
+	"strings"
+)
+
+var (
+	keycloakRealmKeystoreRsaAlgorithm = []string{"RS256", "RS384", "RS512", "PS256", "PS384", "PS512"}
+	keycloakRealmKeystoreRsaSize      = []int{1024, 2048, 4096}
+)
+
+func resourceKeycloakRealmKeystoreRsa() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKeycloakRealmKeystoreRsaCreate,
+		Read:   resourceKeycloakRealmKeystoreRsaRead,
+		Update: resourceKeycloakRealmKeystoreRsaUpdate,
+		Delete: resourceKeycloakRealmKeystoreRsaDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceKeycloakRealmKeystoreRsaImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Display name of provider when linked in admin console.",
+			},
+			"realm_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"active": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Set if the keys can be used for signing",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Set if the keys are enabled",
+			},
+			"priority": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     0,
+				Description: "Priority for the provider",
+			},
+			"algorithm": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(keycloakRealmKeystoreRsaAlgorithm, false),
+				Default:      "RS256",
+				Description:  "Intended algorithm for the key",
+			},
+			"key_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntInSlice(keycloakRealmKeystoreRsaSize),
+				Default:      2048,
+				Description:  "Size for the generated keys",
+			},
+			"private_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Private RSA Key encoded in PEM format",
+			},
+			"certificate": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "X509 Certificate encoded in PEM format",
+			},
+			"disable_read": {
```

Comment:
we probably don't want to add attributes specifically for discarding configuration drift, there are other ways we can deal with that.

it looks like `private_key` comes back from keycloak as `**********` - so we'll probably want to catch this and not set this value in state when reading.  the state value should contain the private key submitted by the user.  this will allow us to detect changes to this attribute appropriately.